app/src/main/java/com/babylon/wallet/android/presentation/transaction/analysis/TransactionAnalysisDelegate.kt

```diff
@@ -67,7 +67,7 @@ class TransactionAnalysisDelegate @Inject constructor(
             }.then { transactionToReview ->
                 _state.update { it.copy(transactionManifestData = TransactionManifestData.from(transactionToReview.transactionManifest)) }
 
-                val manifestSummary = transactionToReview.transactionManifest.summary
+                val manifestSummary = transactionToReview.transactionManifest.summary!!
```

Comment:
As mentioned in previous comments, the transactionManifest returned from analyzeTransaction is assured to have valid summary, probably a better handling would be to return the valid summary along with the manifest in the `transactionToReview`, will potentially make the fix when analyze_v2 will be put in place.

Reply:
For convenience, in `TransactionToReview`, besides the non-optional `ManifestSummary` we can also pass the `Message` as we need it for signing. By the way, this would also reduce the uniffi calls.

Reply:
>For convenience, in TransactionToReview, besides the non-optional ManifestSummary we can also pass the Message as we need it for signing. By the way, this would also reduce the uniffi calls.

+1